// node-rfc/lib/sapnwrfc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RFC_UNIT_STATE = exports.RfcLoggingLevel = exports.RfcLoggingClass = exports.RfcParameterDirection = exports.EnumTrace = exports.EnumSncQop = exports.RFC_RC = void 0;
var RFC_RC;
(function (RFC_RC) {
    RFC_RC[RFC_RC["RFC_OK"] = 0] = "RFC_OK";
    RFC_RC[RFC_RC["RFC_COMMUNICATION_FAILURE"] = 1] = "RFC_COMMUNICATION_FAILURE";
    RFC_RC[RFC_RC["RFC_LOGON_FAILURE"] = 2] = "RFC_LOGON_FAILURE";
    RFC_RC[RFC_RC["RFC_ABAP_RUNTIME_FAILURE"] = 3] = "RFC_ABAP_RUNTIME_FAILURE";
    RFC_RC[RFC_RC["RFC_ABAP_MESSAGE"] = 4] = "RFC_ABAP_MESSAGE";
    RFC_RC[RFC_RC["RFC_ABAP_EXCEPTION"] = 5] = "RFC_ABAP_EXCEPTION";
    RFC_RC[RFC_RC["RFC_CLOSED"] = 6] = "RFC_CLOSED";
    RFC_RC[RFC_RC["RFC_CANCELED"] = 7] = "RFC_CANCELED";
    RFC_RC[RFC_RC["RFC_TIMEOUT"] = 8] = "RFC_TIMEOUT";
    RFC_RC[RFC_RC["RFC_MEMORY_INSUFFICIENT"] = 9] = "RFC_MEMORY_INSUFFICIENT";
    RFC_RC[RFC_RC["RFC_VERSION_MISMATCH"] = 10] = "RFC_VERSION_MISMATCH";
    RFC_RC[RFC_RC["RFC_INVALID_PROTOCOL"] = 11] = "RFC_INVALID_PROTOCOL";
    RFC_RC[RFC_RC["RFC_SERIALIZATION_FAILURE"] = 12] = "RFC_SERIALIZATION_FAILURE";
    RFC_RC[RFC_RC["RFC_INVALID_HANDLE"] = 13] = "RFC_INVALID_HANDLE";
    RFC_RC[RFC_RC["RFC_RETRY"] = 14] = "RFC_RETRY";
    RFC_RC[RFC_RC["RFC_EXTERNAL_FAILURE"] = 15] = "RFC_EXTERNAL_FAILURE";
    RFC_RC[RFC_RC["RFC_EXECUTED"] = 16] = "RFC_EXECUTED";
    RFC_RC[RFC_RC["RFC_NOT_FOUND"] = 17] = "RFC_NOT_FOUND";
    RFC_RC[RFC_RC["RFC_NOT_SUPPORTED"] = 18] = "RFC_NOT_SUPPORTED";
    RFC_RC[RFC_RC["RFC_ILLEGAL_STATE"] = 19] = "RFC_ILLEGAL_STATE";
    RFC_RC[RFC_RC["RFC_INVALID_PARAMETER"] = 20] = "RFC_INVALID_PARAMETER";
    RFC_RC[RFC_RC["RFC_CODEPAGE_CONVERSION_FAILURE"] = 21] = "RFC_CODEPAGE_CONVERSION_FAILURE";
    RFC_RC[RFC_RC["RFC_CONVERSION_FAILURE"] = 22] = "RFC_CONVERSION_FAILURE";
    RFC_RC[RFC_RC["RFC_BUFFER_TOO_SMALL"] = 23] = "RFC_BUFFER_TOO_SMALL";
    RFC_RC[RFC_RC["RFC_TABLE_MOVE_BOF"] = 24] = "RFC_TABLE_MOVE_BOF";
    RFC_RC[RFC_RC["RFC_TABLE_MOVE_EOF"] = 25] = "RFC_TABLE_MOVE_EOF";
    RFC_RC[RFC_RC["RFC_START_SAPGUI_FAILURE"] = 26] = "RFC_START_SAPGUI_FAILURE";
    RFC_RC[RFC_RC["RFC_ABAP_CLASS_EXCEPTION"] = 27] = "RFC_ABAP_CLASS_EXCEPTION";
    RFC_RC[RFC_RC["RFC_UNKNOWN_ERROR"] = 28] = "RFC_UNKNOWN_ERROR";
    RFC_RC[RFC_RC["RFC_AUTHORIZATION_FAILURE"] = 29] = "RFC_AUTHORIZATION_FAILURE";
    RFC_RC[RFC_RC["RFC_AUTHENTICATION_FAILURE"] = 30] = "RFC_AUTHENTICATION_FAILURE";
    RFC_RC[RFC_RC["RFC_CRYPTOLIB_FAILURE"] = 31] = "RFC_CRYPTOLIB_FAILURE";
    RFC_RC[RFC_RC["RFC_IO_FAILURE"] = 32] = "RFC_IO_FAILURE";
    RFC_RC[RFC_RC["RFC_LOCKING_FAILURE"] = 33] = "RFC_LOCKING_FAILURE";
})(RFC_RC || (exports.RFC_RC = RFC_RC = {}));
var EnumSncQop;
(function (EnumSncQop) {
    EnumSncQop["DigSig"] = "1";
    EnumSncQop["DigSigEnc"] = "2";
    EnumSncQop["DigSigEncUserAuth"] = "3";
    EnumSncQop["BackendDefault"] = "8";
    EnumSncQop["Maximum"] = "9";
})(EnumSncQop || (exports.EnumSncQop = EnumSncQop = {}));
var EnumTrace;
(function (EnumTrace) {
    EnumTrace["Off"] = "0";
    EnumTrace["Brief"] = "1";
    EnumTrace["Verbose"] = "2";
    EnumTrace["Full"] = "3";
})(EnumTrace || (exports.EnumTrace = EnumTrace = {}));
var RfcParameterDirection;
(function (RfcParameterDirection) {
    RfcParameterDirection[RfcParameterDirection["RFC_IMPORT"] = 1] = "RFC_IMPORT";
    RfcParameterDirection[RfcParameterDirection["RFC_EXPORT"] = 2] = "RFC_EXPORT";
    RfcParameterDirection[RfcParameterDirection["RFC_CHANGING"] = 3] = "RFC_CHANGING";
    RfcParameterDirection[RfcParameterDirection["RFC_TABLES"] = 7] = "RFC_TABLES";
})(RfcParameterDirection || (exports.RfcParameterDirection = RfcParameterDirection = {}));
var RfcLoggingClass;
(function (RfcLoggingClass) {
    RfcLoggingClass[RfcLoggingClass["client"] = 0] = "client";
    RfcLoggingClass[RfcLoggingClass["pool"] = 1] = "pool";
    RfcLoggingClass[RfcLoggingClass["server"] = 2] = "server";
    RfcLoggingClass[RfcLoggingClass["throughput"] = 3] = "throughput";
    RfcLoggingClass[RfcLoggingClass["nwrfc"] = 4] = "nwrfc";
    RfcLoggingClass[RfcLoggingClass["addon"] = 5] = "addon";
})(RfcLoggingClass || (exports.RfcLoggingClass = RfcLoggingClass = {}));
var RfcLoggingLevel;
(function (RfcLoggingLevel) {
    RfcLoggingLevel[RfcLoggingLevel["none"] = 0] = "none";
    RfcLoggingLevel[RfcLoggingLevel["fatal"] = 1] = "fatal";
    RfcLoggingLevel[RfcLoggingLevel["error"] = 2] = "error";
    RfcLoggingLevel[RfcLoggingLevel["warning"] = 3] = "warning";
    RfcLoggingLevel[RfcLoggingLevel["info"] = 4] = "info";
    RfcLoggingLevel[RfcLoggingLevel["debug"] = 5] = "debug";
    RfcLoggingLevel[RfcLoggingLevel["all"] = 6] = "all";
})(RfcLoggingLevel || (exports.RfcLoggingLevel = RfcLoggingLevel = {}));
var RFC_UNIT_STATE;
(function (RFC_UNIT_STATE) {
    RFC_UNIT_STATE[RFC_UNIT_STATE["RFC_UNIT_NOT_FOUND"] = 0] = "RFC_UNIT_NOT_FOUND";
    RFC_UNIT_STATE[RFC_UNIT_STATE["RFC_UNIT_IN_PROCESS"] = 1] = "RFC_UNIT_IN_PROCESS";
    RFC_UNIT_STATE[RFC_UNIT_STATE["RFC_UNIT_COMMITTED"] = 2] = "RFC_UNIT_COMMITTED";
    RFC_UNIT_STATE[RFC_UNIT_STATE["RFC_UNIT_ROLLED_BACK"] = 3] = "RFC_UNIT_ROLLED_BACK";
    RFC_UNIT_STATE[RFC_UNIT_STATE["RFC_UNIT_CONFIRMED"] = 4] = "RFC_UNIT_CONFIRMED";
})(RFC_UNIT_STATE || (exports.RFC_UNIT_STATE = RFC_UNIT_STATE = {}));
